core/source/Array.mint

```diff
@@ -15,6 +15,17 @@ module Array {
     Array.find(array, function) != Maybe.Nothing
   }
 
+  /*
+  Returns `true` if all items in the array matche the predicate function,
+    `false` otherwise.
+
+    Array.all([1, 2, 3, 4], (number : Number) { number % 2 == 0 }) == false
+    Array.all(["hello", "mint"], (str : String) { String.size(str) > 3 }) == true
+  */
+  fun all (array : Array(item), function : Function(item, Bool)) : Bool {
+    reduce(array, true, (memo : Bool, val : item) { memo && function(val) })
```

Comment:
Design decision: does the empty array always satisfy "all" or never satisfy "all"? In Crystal the answer is that it always satisfies "all" (ref: https://play.crystal-lang.org/#/r/htfj) so I made this behavior match.

Reply:
It should return true for empty arrays: https://en.wikipedia.org/wiki/Vacuous_truth#In_computer_programming :+1: